src/mono/mono.proj

```diff
@@ -39,8 +39,17 @@
     <MonoCxxCompiler Condition="'$(MonoCCompiler)' == 'gcc'">g++</MonoCxxCompiler>
   </PropertyGroup>
 
+  <Target Name="CheckMonoEnv">
+	<!-- EMSDK_PATH is set by CI or we set it to the location of a local copy -->
+    <Exec Command="make -C $(MonoProjectRoot)wasm check-provision-wasm" Condition="'$(TargetsBrowser)' == 'true' and '$(EMSDK_PATH)' == ''" />
+
+    <PropertyGroup Condition="'$(TargetsBrowser)' == 'true'">
+      <EMSDK_PATH Condition="'$(EMSDK_PATH)' == ''">$(MonoProjectRoot)wasm/emsdk</EMSDK_PATH>
+    </PropertyGroup>
```

Comment:
This won't work as is since we can't propagate the EMSDK_PATH value from here to the libraries build, so it would only work for the mono runtime build.

In my opinion what we should do instead is print out a message if EMSDK_PATH isn't set telling the user to either set it, or run the `provision-wasm` script which would download emscripten and print out `EMSDK_PATH=.../src/mono/wasm/emsdk` which the user can the copy and rerun the build.